test/strategy.test.js

```diff
@@ -277,3 +284,42 @@ describe('#authenticate', function (){
     });
   });
 });
+
+describe('#Issuer', function() {
+  it('returns a discover function and well known config object', function() {
+    expect(Issuer.discover).to.be.a('function');
+    expect(Issuer._well_known_config).to.be.an('object');
+  });
+
+  describe('#discover', function() {
+    context('when a url is passed', function() {
+      it('returns a Client contructor and well known config object', function() {
+        Issuer.discover('localhost:5000/').then(function(issuer) {
+          expect(issuer.Client).to.be.a('function');
```

Comment:
you need to return this promise so the test knows to wait for the epects to resolve

Reply:
we are using chai-as-promised. we can break this by: 
```
describe('#discover', function() {
    context('when a url is passed', function() {
      it('returns a Client contructor and well known config object', function() {
        Issuer.discover('localhost:5000/').then(function(issuer) {
          expect(issuer.Client).to.be.a('string');
        });
      });
    });
```
results in: 
```
 #Issuer
    ✓ returns a discover function and well known config object
    #discover
      when a url is passed
        ✓ returns a Client contructor and well known config object
Unhandled rejection AssertionError: expected [Function: Client] to be a string
```